Assets/Scripts/Audio/AudioController.cs

```diff
@@ -169,6 +169,19 @@ public void SetMusicVolume(int volume)
 		onMusicVolumeChange(this.musicVolume);
 	}
 
+    //BP Does the sound already have a channel (ie an audiosource on 'Audio' gameObject)?
+    bool ClipHasChannel(AudioFile file)
+    {
+        foreach(AudioSource channel in channels.Values)
```

Comment:
Instead of doing an O(n) lookup each time to check, think it would be more efficient to use a HashSet to track which AudioClips are active: https://msdn.microsoft.com/en-us/library/bb359438(v=vs.110).aspx. This provides O(1) speed to lookup each time